queue/abstract.lua

```diff
@@ -352,6 +359,7 @@ function method._on_consumer_disconnect()
         box.space._queue_consumers:delete{ waiter[1], waiter[2] }
         local cond = conds[waiter[2]]
         if cond then
+            releasing_sessions[waiter[2]] = true
             cond:signal(waiter[2])
```

Comment:
The reason why we do it here is to pass the fiber back to the fiber pool. I think fiber.cancel() perfectly fits this need. (NB: don't forget to free the condition variable.)